src/main/java/lotto/util/InputValidator.java

```diff
@@ -0,0 +1,108 @@
+    private static final int MIN_LOTTO_NUMBER = 1;
+    private static final int MAX_LOTTO_NUMBER = 45;
+    private static final int MIN_PURCHASE_AMOUNT = 1000;
+    private static final int MAX_PURCHASE_AMOUNT = 10000000;
+    private static final String NEED_NUMBER_IN_RANGE = "구매 금액은 1000이상 10000000이하여야 합니다.";
+
+
+    @Override
+    public void checkIfEmpty(String input){
+        if (input.isEmpty()) {
+            throw new InputErrorException(ErrorType.NEED_NOT_EMPTY);
+        }
+        if (input.replaceAll(WHITE_SPACE_REGEX, EMPTY_STRING).isEmpty()) {
+            throw new InputErrorException(ErrorType.NEED_NOT_EMPTY);
+        }
+    }
+
+    @Override
+    public void checkIfEmpty(List<Integer> input){
+        if (input.isEmpty()) {
+            throw new InputErrorException(ErrorType.NEED_NOT_EMPTY);
+        }
+    }
+
+
+    @Override
+    public void checkIfDigit(String input){
+        for (int i = 0; i < input.length(); i++) {
+            if (!Character.isDigit(input.charAt(i))) {
+                throw new InputErrorException(ErrorType.NEED_NUMBER);
+            }
+        }
+    }
+
+    @Override
+    public void validateOnlyDigit(List<String> lottoNumbers){
+        lottoNumbers.forEach(lottoNumber -> {
+            checkIfEmpty(lottoNumber);
+            checkIfDigit(lottoNumber);
+        });
+    }
+
+
+    @Override
+    public void validateOnlyDigit(String input){
+        checkIfEmpty(input);
+        checkIfDigit(input);
+    }
+
+    @Override
+    public void checkValidRange(Lotto lotto){
+        lotto.getLottoNumbers().forEach(lottoNumber -> {
+            if (lottoNumber < MIN_LOTTO_NUMBER || lottoNumber > MAX_LOTTO_NUMBER) {
+                throw new InputErrorException(ErrorType.NEED_NUMBER_IN_RANGE);
+            }
+        });
+
+    }
+
+
+    @Override
+    public void checkValidRange(Integer bonusNumber){
+        if (bonusNumber < MIN_LOTTO_NUMBER || bonusNumber > MAX_LOTTO_NUMBER) {
+            throw new InputErrorException(ErrorType.NEED_NUMBER_IN_RANGE);
+        }
+    }
+
+    @Override
+    public void checkValidRange(int lottoCount){
+        if (lottoCount < MIN_PURCHASE_AMOUNT || lottoCount > MAX_PURCHASE_AMOUNT) {
+            throw new InputErrorException(NEED_NUMBER_IN_RANGE);
+        }
+    }
+
+
+    @Override
+    public void checkDuplicate(List<Integer> numbers){
+        Set<Integer> distinctLottoNumbers = new HashSet<>();
+
+        for (Integer lottoNumber : numbers) {
+            if (!distinctLottoNumbers.add(lottoNumber)) {
+                throw new InputErrorException(ErrorType.NEED_DISTINCT_NUMBER);
+            }
+        }
+    }
+
+    @Override
+    public void checkDuplicate(Integer bonusNumber, Lotto lotto) {
+        lotto.getLottoNumbers().forEach(lottoNumber -> {
+            if (lottoNumber.equals(bonusNumber)) {
+                throw new InputErrorException(ErrorType.NEED_DISTINCT_NUMBER);
+            }
+        });
```

Comment:
반복하기보다 if문 하나로 가능할 것 같아요!
첨언하자면, lotto의 numbers를 꺼내와서 검사하기보다, 로또 객체에게 "너 이 숫자 가지고있어?"하고 물어보는 형태가 더 객체지향적일 듯 하다고 생각합니다!
```suggestion
        if (lotto.getLottoNumbers().contains(bonusNumber)) {
            throw new InputErrorException(ErrorType.NEED_DISTINCT_NUMBER);
        }
```

Reply:
네 말씀하신 내용 동의합니다! 이번 미션 공통 피드백에서도 getter로 값을 받아서 외부에서 로직을 수행하는 것보다 객체 스스로 메시지에 따라 행동하도록 해야한다고 하더라고요. 그래서 로또 객체가 직접 bonusNumber를 받아서 검사하는 게 더 객체지향적이라는 생각이 듭니다.